src/scripts/main.js

```diff
@@ -1,3 +1,193 @@
+    <label>Office:
+      <select name="office" data-qa="office" required>
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age: <input name="age" type="number"
+      data-qa="age" required>
+    </label>
+    <label>
+      Salary: <input name="salary" type="number"
+        data-qa="salary" required>
+    </label>
+    <button name="button" type="submit">Save to table</button>
+  </form>
+`,
+);
+
+const form = document.querySelector('.new-employee-form');
+
+function createNotification(type, description) {
+  document.body.insertAdjacentHTML(
+    'afterbegin',
+    `
+    <div class="notification ${type}" data-qa="notification">
+      <span class="title">${type}</span>
+      <p>${description}</p>
+    </div>
+  `,
+  );
+
+  const message = document.querySelector('.notification');
+
+  setTimeout(() => {
+    message.remove();
+  }, 3000);
+}
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const nameEmp = form.name.value;
+  const position = form.position.value;
+  const office = form.office.value;
+  const age = Number(form.age.value);
+  const salary = Number(form.salary.value);
+
+  let errorMessage = '';
+  let isValid = true;
+
+  if (nameEmp.length < 4) {
+    errorMessage += 'Name must be at least 4 characters long';
+    isValid = false;
+  }
+
+  if (age < 18 || age > 90) {
+    errorMessage += 'Age must be between 18 and 90 years';
+    isValid = false;
+  }
+
+  if (!isValid) {
+    createNotification('error', errorMessage);
+
+    return;
+  }
+
+  tbody.insertAdjacentHTML(
+    'beforeend',
+    `
+    <tr>
+      <td>${nameEmp}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${`$${salary.toLocaleString('en-US')}`}</td>
+    </tr>
+  `,
+  );
+
+  rows.push(tbody.lastElementChild);
+
+  createNotification('success', 'Success! Employee added to the table');
+  form.reset();
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const target = e.target;
+
+  if (target.tagName === 'TD') {
+    const currentText = target.textContent.trim();
+    const input = document.createElement('input');
+
+    input.value = currentText;
+    input.className = 'cell-input';
+    target.textContent = '';
+    target.appendChild(input);
+    input.focus();
+
+    input.addEventListener('blur', () => {
+      const newValue = input.value.trim();
+
+      if (newValue === '') {
+        target.textContent = currentText;
+      } else {
+        target.textContent = newValue;
+      }
+    });
+
+    input.addEventListener('keypress', (ev) => {
+      if (ev.key === 'Enter') {
+        const newValue = input.value.trim();
+
+        if (newValue === '') {
+          target.textContent = currentText;
+        } else {
+          target.textContent = newValue;
+        }
```

Comment:
Similar to the blur event, when the Enter key is pressed, the code reverts to the previous text if the input is empty. It might be beneficial to notify the user about this behavior.